submissions/krylenger/memory-pair-game/app.js

```diff
@@ -0,0 +1,244 @@
+    }
+    
+}
+
+const createNewCard = (card, cardNumber) => {
+    console.log(cardNumber);
+    const newCard = document.createElement('div');
+    const newCardFlipper = document.createElement('div');
+    const newCardFlipperFront = document.createElement('div');
+    const newCardFlipperBack = document.createElement('div');
+    if (card.musicCard) {
+        newCard.classList.add('card', `card--${cardNumber}`, 'musicCard')
+        newCardFlipperBack.classList.add(`card--music-img`);
+    } else {
+        newCard.classList.add('card', `card--${cardNumber}`);
+        newCardFlipperBack.classList.add(`card--${card.text}-img`);
+    }
+    newCardFlipper.classList.add('card__flipper');
+    newCardFlipper.setAttribute('ontouchstart', "this.classList.toggle('card__flipper--flip');")
+    styleNewCardFlipperFront(newCardFlipperFront, card, cardNumber);
+    newCardFlipperBack.classList.add('card__flipper-back');
+    newCard.appendChild(newCardFlipper);
+    newCardFlipper.appendChild(newCardFlipperFront);
+    newCardFlipper.appendChild(newCardFlipperBack);
+    return newCard;
+}
+
+const shuffleArr = (arr) => {
+    return arr.sort(() => 0.5 - Math.random());
+}
+
+const loadCards = (cardsArr) => {
+    const shuffledCards = shuffleArr(cardsArr);
+    const fragment = document.createDocumentFragment();
+    shuffledCards.forEach((card, index) => {
+        const newCard = createNewCard(card, index);
+        fragment.appendChild(newCard);
+    })
+    mainGridContainer.appendChild(fragment);
+}
+
+const recreateMainGridContainer = () => {
+    mainGridContainer.remove();
+    mainGridContainer = document.createElement('div');
+    mainGridContainer.classList.add('main__grid-container');
+    main.appendChild(mainGridContainer);
+}
+
+const resetGame = () => {
+    recreateMainGridContainer();
+    loadCards(cards);
+    guessedCards = [];
+    playedSongs = [];
+    handleClick();
+}
+
+const handlePlayingCards = (cardStatus) => {
+    const playingNow = document.querySelectorAll(".playingNow");
+    playingNow.forEach((card) => {
+        card.classList.remove("playingNow");
+        if (cardStatus === 'guessed') {
+            setTimeout(() => {
+                card.parentNode.parentNode.classList.add('card--background-green');
+            }, 300)
+        }; 
+        setTimeout(() => {
+            card.parentNode.classList.toggle(cardStatus);
+        }, 1000)
+    });
+}
+
+
+const handleGameOver = (guessedCards) => {
+    if (guessedCards === 10) {
+        setTimeout(() => {
+            alert(`Congratulations! Let's play once more!`);
+            resetGame(guessedCards);
+        }, 1000)
+    }
+}
+
+const stopPreviousSong = (playedSongs) => {
+    let previousSong = playedSongs[playedSongs.length - 1];
+    previousSong.pause();
+    previousSong.currentTime = 0;
+}
+
+const handleMusicCard = (target) => {
+    if (target.parentNode.parentNode.classList.contains('musicCard')) {
```

Comment:
use `target.closest('someSelector')`, please because in your approach you strongly depends on your layout

Reply:
✅ Done